WordPress/wordpress-develop
# Allow the style property of block.json to be an array and add support for object-based block styles

## What problem does this PR solve?

In https://github.com/WordPress/gutenberg/pull/34180 we moved block CSS to `blockJson.supports.__experimentalStyle`. This PR moves it under `blockJson.style` at the root level. The goal is to only have a single location where styles are defined.

Block.json **before** this PR:

```js
{
	"supports":  {
		"__experimentalStyle": {
			"border": {			
				"color": {
					"text": "#fff",
					"background": "#32373c"
				}
			}
		}
	},
	"style":  "wp-block-button"
}
```

Block.json **after** this PR:

```js
{
	"style":  [ "wp-block-button", {
		"border": {			
			"color": {
				"text": "#fff",
				"background": "#32373c"
			}
		}
	}]
}
```

This PR is an alternative to https://github.com/WordPress/gutenberg/pull/41656 

## Other considerations

This is a breaking change – any existing code that assumes `style` to be a single item or an array of strings will throw a notice at best.

## Test plan

1. Apply this PR to your local wordpress-develop
2. Run `npm run build`
3. Apply https://github.com/WordPress/gutenberg/pull/41873 to your local Gutenberg installation
4. Build Gutenberg
5. Switch themes to TwentyTwentyTwo
6. Create a new page
7. Insert a button, give it some text
8. Save it, view it on the frontend
9. Confirm the buttons have a large padding at the bottom

cc @getdave @scruffian @draganescu 

Trac ticket: https://core.trac.wordpress.org/ticket/56094
